multiplex.go

```diff
@@ -171,7 +199,10 @@ func (mp *Multiplex) CloseChan() <-chan struct{} {
 }
 
 func (mp *Multiplex) sendMsg(timeout, cancel <-chan struct{}, header uint64, data []byte) error {
-	buf := pool.Get(len(data) + 20)
+	buf, err := mp.pool.Get(len(data) + 20)
```

Comment:
The problem here is that returning an error might not lead to a connection failure. For example, in `NewStream` it will just cause us to return an error and a `nil` stream, but not do anything with the stream that was already created.
This is not something that this PR introduced though... Still wondering if we maybe should _always_ kill the connection when an allocation fails.